app/auth.py

```diff
@@ -1,11 +1,21 @@
-from flask import Blueprint, render_template, redirect, url_for, request, flash
+from flask import Blueprint, render_template, redirect, url_for, request, flash, current_app, session
 from werkzeug.security import generate_password_hash, check_password_hash
 from flask_login import login_user, login_required, logout_user
+from flask_principal import identity_changed, Identity, AnonymousIdentity
 from .models import User
 from . import db
 
 auth = Blueprint('auth', __name__)
 
+'''  
+TODO:
+admin_permission = Permission(RoleNeed('admin'))
+@auth.route('/admin')
+@admin_permission.require()
+def do_admin_index():
+    return render_template('admin.html')
+'''      
+
```

Comment:
Until we knock out this TODO, @stuart909 will have to manually associate his testing credentials with 'admin' to click through any views protected with the `@admin_permission.require()` decorator.